apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -333,6 +335,40 @@ public EmulatorDetection EmulatorDetection
             }
         }
 
+        /// <summary>
+        /// Other known clients may set an additional value for the x-goog-api-client version header. This
+        /// array contains the only valid values.
+        /// </summary>
+        private static readonly string[] s_allowedVersionHeaders = { "efcore" };
```

Comment:
I don't think we should check this. If values here are relevant, then it should be the API the one to check not us.
And anyway, anyone using Spanner.Data can now set "efcore" themselves (without actually using the EF Core library) and it would muddy whatever is being tracked with this.
The header is in the end a good faith kind of identifier, for Spanner and the rest of the libraries.